library/std/src/os/windows/net/listener.rs

```diff
@@ -0,0 +1,99 @@
+use crate::os::windows::io::{AsRawSocket, FromRawSocket, IntoRawSocket, RawSocket};
+use crate::os::windows::net::{SocketAddr, UnixStream, not_cvt};
+use crate::path::Path;
+use crate::sys::net::Socket;
+use crate::{io, mem};
+#[cfg(windows)]
+use crate::{
+    os::windows::net::socketaddr_un,
+    sys::{
+        c::{AF_UNIX, SOCK_STREAM, SOCKADDR_UN, bind, getsockname, listen},
+        winsock::startup,
+    },
+};
+pub struct UnixListener(pub Socket);
```

Comment:
```suggestion
pub struct UnixListener(Socket);
```

The public API also shouldn't expose internal fields like this.